### ISSUE TITLE
Randomly thows errors

### ISSUE THREAD

**Issue by buger** [mixpanel/mixpanel-ruby#73]
It happen from time to time, probably because some API issues, but definitely this case not handled:

```
irb(main):006:0> tracker.track(2, "Viewed people search")
NoMethodError: undefined method `[]' for nil:NilClass
	from /usr/local/bundle/gems/mixpanel-ruby-2.0.1/lib/mixpanel-ruby/consumer.rb:106:in `send!'
	from org/jruby/RubyMethod.java:124:in `call'
	from /usr/local/bundle/gems/mixpanel-ruby-2.0.1/lib/mixpanel-ruby/events.rb:70:in `track'
	from /usr/local/bundle/gems/mixpanel-ruby-2.0.1/lib/mixpanel-ruby/tracker.rb:72:in `track'
	from (irb):6:in `evaluate'
	from org/jruby/RubyKernel.java:1107:in `eval'
	from org/jruby/RubyKernel.java:1507:in `loop'
	from org/jruby/RubyKernel.java:1270:in `catch'
	from org/jruby/RubyKernel.java:1270:in `catch'
	from /opt/jruby/bin/irb:13:in `(root)'
```

**Comment by buger** [comment]
I made some debugging (simple monkey patch: https://gist.github.com/buger/7eb768b352785cc34b4d)
And here what it gets from mixpanel server: 

irb(main):185:0> tracker.track(2, "Viewed people search")
{:uri=>"/track", :endpoint=>"https://api.mixpanel.com/track", :form_data=>{"data"=>"eyJldmVudCI6IlZpZXdlZCBwZW9wbGUgc2VhcmNoIiwicHJvcGVydGllcyI6eyJkaXN0aW5jdF9pZCI6MiwidG9rZW4iOiI0NjUzYzFkNzY1ZGUzMjIyNDQzMWY4ZDhlMTUxMTg4OSIsInRpbWUiOjE0MjgxNzcwOTAsIm1wX2xpYiI6InJ1YnkiLCIkbGliX3ZlcnNpb24iOiIyLjAuMSJ9fQ==", "verbose"=>1}, :code=>"200", :body=>""}
NoMethodError: undefined method `[]' for nil:NilClass
	from /usr/local/bundle/gems/mixpanel-ruby-2.0.1/lib/mixpanel-ruby/consumer.rb:106:in `send!'

**Comment by buger** [comment]
Also seems like mixpanel still tracks this events.

**Comment by zyskowsk** [comment]
Hey @buger!  thanks for the note.  I'm going to look into why we might be spastically returning an empty body.  If this is something that happens regularly we might want to catch this error in the consumer, along with other similar errors, as in #71.

**Comment by zyskowsk** [comment]
addressed in #77
